### PR TITLE
fix: Improve error message for misapplied config

### LINKED ISSUE #133: GCP Credential Issues
### Describe the bug

- When provide a file path to the `CQ_SERVICE_ACCOUNT_KEY_JSON` you get a weird none helpful error
- Documentation is not clear about `CQ_SERVICE_ACCOUNT_KEY_JSON` and `GOOGLE_APPLICATION_CREDENTIALS`

### Expected Behavior

Clarity and ease of use. Customers should be able to at the very least begin debugging themselves. Right now it is impossible 

### Steps to Reproduce

provide a path `CQ_SERVICE_ACCOUNT_KEY_JSON` as the value

### Possible Solution

_No response_

### Provider and CloudQuery version

all?

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by spangenberg**
`CQ_SERVICE_ACCOUNT_KEY_JSON` needs to be a valid JSON and not such a path.

Fixes #133

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -58,6 +60,14 @@ func Configure(logger hclog.Logger, config interface{}) (schema.ClientMeta, erro
...
 	if len(serviceAccountKeyJSON) == 0 {
 		serviceAccountKeyJSON = []byte(os.Getenv(serviceAccountEnvKey))
+		var v map[interface{}]interface{}
+		err := json.Unmarshal(serviceAccountKeyJSON, &v)
+		var syntaxError *json.SyntaxError
+		if errors.As(err, &syntaxError) {
+			return nil, fmt.Errorf("the environment variable %v should contain valid JSON and not just a file path to a JSON file", serviceAccountEnvKey)
+		} else if err != nil {
+			return nil, err
+		}
```
Comment (bbernays) [inline client/client.go:70]:
```suggestion
	if err != nil {
		var syntaxError *json.SyntaxError
		if errors.As(err, &syntaxError) {
			return nil, fmt.Errorf("the environment variable %v should contain valid JSON and not just a file path to a JSON file", serviceAccountEnvKey)
		}
		return nil, err
	}

```

File: client/client.go [hunk 2]
```diff
@@ -58,6 +60,14 @@ func Configure(logger hclog.Logger, config interface{}) (schema.ClientMeta, erro
 	serviceAccountKeyJSON := []byte(providerConfig.ServiceAccountKeyJSON)
 	if len(serviceAccountKeyJSON) == 0 {
 		serviceAccountKeyJSON = []byte(os.Getenv(serviceAccountEnvKey))
+		var v map[interface{}]interface{}
```
Comment (bbernays) [inline client/client.go:63]: I am getting an error when unmarshalling with `map[interface{}]interface{}` but it works with `map[string]interface{}`

**Review by bbernays — COMMENTED** [review]
if err is nil

File: client/client.go [hunk 3]
```diff
@@ -58,6 +60,14 @@ func Configure(logger hclog.Logger, config interface{}) (schema.ClientMeta, erro
...
+		err := json.Unmarshal(serviceAccountKeyJSON, &v)
+		var syntaxError *json.SyntaxError
+		if errors.As(err, &syntaxError) {
+			return nil, fmt.Errorf("the environment variable %v should contain valid JSON and not just a file path to a JSON file", serviceAccountEnvKey)
```
Comment (disq) [inline client/client.go:67]: I'm not sure about assuming the value being a file path, could be confusing if it's just an invalid json.

Comment (bbernays) [reply client/client.go:67]: Either we can have the error message say `the environment variable %v should contain valid JSON object` or we could check to see if the path exists and if the path exists then we can let the customer know that they shouldn't specify a path

Comment (spangenberg) [reply client/client.go:68]: Was thinking about that, but also didn't want to overengineer the solution. I thought the most common issue is to supply a path and to state that. I'm fine with either, but when we check that the value is a path we could also just resolve it and feed the content of the path.

Comment (bbernays) [reply client/client.go]: I am fine with just the generic error

Comment (spangenberg) [reply client/client.go]: @bbernays fixed it.

**Review by bbernays — APPROVED** [review]
LGTM

**Comment by yevgenypats** [comment]
Apparently, this breaks the provider as it doesn't work anymore without CQ_SERVICE_ACCOUNT_KEY_JSON. Pushing a fix shortly together with tests :) 

cc @spangenberg @disq @bbernays

**Comment by yevgenypats** [comment]
Also I think it doesn't actually solve the issue at all (at least mostly) as it does the json check only in case it is passed via `CQ_SERVICE_ACCOUNT_KEY_JSON`

**Pull request merged**